### PR TITLE
Better way to retrieve the database connection.

### PR DESCRIPTION
**Pull request by casanasvictor**
TL;DR uses Illuminate\Support\Facades\DB to create the connection

---

# Reason
The reason behind this pr is when using dynamic databases like on multi-tenant applications often the database name is null and set dynamically depending on the request so getting the db connection out of the container like:
** $this->container['db']->connection($config['connection'] ?? null) ** will always set the connection using the wrong configuration and no the dynamic one

---

# How to Reproduce
on the config/database.php

create a connection like this:

'connections' => [

        'tenant' => [
            'driver' => 'mysql',
            'url' => env('DATABASE_URL'),
            'host' => env('DB_HOST', '127.0.0.1'),
            'port' => env('DB_PORT', '3306'),
            'database' => null, 
            'username' => env('DB_USERNAME', 'forge'),
            'password' => env('DB_PASSWORD', ''),
            'unix_socket' => env('DB_SOCKET', ''),
            'charset' => 'utf8mb4',
            'collation' => 'utf8mb4_unicode_ci',
            'prefix' => '',
            'prefix_indexes' => true,
            'strict' => true,
            'engine' => null,
            'options' => extension_loaded('pdo_mysql') ? array_filter([
                PDO::MYSQL_ATTR_SSL_CA => env('MYSQL_ATTR_SSL_CA'),
                'sticky' => false,
            ]) : [],
        ],
        
        
in order to simulate a multi-tenant app change the config files in any route, request, etc like this:

config([
            'database.connections.tenant.database' => 'any_database',
        ]);
  DB::purge('tenant');
  
  Finally use Laravel pennant as you like and you will notice an error:
  
 ** Call to a member function prepare() on null **

### CODE REVIEW

**Comment by timacdonald** [comment]
Hi @casanasvictor, thank you for the suggestion and the PR. I've created a PR with a slightly different approach https://github.com/laravel/pennant/pull/43
